sdk/search/azure-search/src/main/java/com/azure/search/SearchIndexClientBuilder.java

```diff
@@ -0,0 +1,265 @@
+        clientName = properties.getOrDefault(NAME, "UnknownName");
+        clientVersion = properties.getOrDefault(VERSION, "UnknownVersion");
+        headers = new HttpHeaders()
+            .put(ECHO_REQUEST_ID_HEADER, "true");
+    }
+
+    /**
+     * Sets the api version to work against
+     *
+     * @param apiVersion api version
+     * @return the updated SearchIndexClientBuilder object
+     */
+    public SearchIndexClientBuilder apiVersion(SearchServiceVersion apiVersion) {
+        if (apiVersion == null) {
+            throw logger.logExceptionAsError(new IllegalArgumentException("Invalid apiVersion"));
+        }
+        this.apiVersion = apiVersion;
+        return this;
+    }
+
+    /**
+     *  Returns the list of policies configured for this builder.
+     *
+     *  @return List of HttpPipelinePolicy
+     *  */
+    public List<HttpPipelinePolicy> getPolicies() {
+        return this.policies;
+    }
+
+    /**
+     * Sets the Azure Cognitive Search service endpoint
+     *
+     * @param endpoint the endpoint URL to the Azure Cognitive Search service
+     * @return the updated SearchIndexClientBuilder object
+     * @throws IllegalArgumentException on invalid service endpoint
+     */
+    public SearchIndexClientBuilder endpoint(String endpoint) {
+        try {
+            new URL(endpoint);
+        } catch (MalformedURLException ex) {
+            throw logger.logExceptionAsWarning(new IllegalArgumentException("'endpoint' must be a valid URL"));
+        }
+        this.endpoint = endpoint;
+        return this;
+    }
+
+    /**
+     * Sets the index name
+     *
+     * @param indexName name of the index
+     * @return the updated SearchIndexClientBuilder object
+     */
+    public SearchIndexClientBuilder indexName(String indexName) {
+        if (StringUtils.isBlank(indexName)) {
+            throw logger.logExceptionAsError(new IllegalArgumentException("Invalid indexName"));
+        }
+        this.indexName = indexName;
+        return this;
+    }
+
+    /**
+     * Set the http client (optional). If this is not set, a default httpClient will be created
+     *
+     * @param httpClient value of httpClient
+     * @return the updated SearchIndexClientBuilder object
+     */
+    public SearchIndexClientBuilder httpClient(HttpClient httpClient) {
+        this.httpClient = httpClient;
+        return this;
+    }
+
+    /**
+     * Sets the api key to use for request authentication.
+     *
+     * @param searchApiKeyCredential api key for request authentication
+     * @return the updated SearchIndexClientBuilder object
+     * @throws IllegalArgumentException when the api key is empty
+     * @throws IllegalArgumentException when the api key is empty
+     */
+    public SearchIndexClientBuilder credential(SearchApiKeyCredential searchApiKeyCredential) {
+        if (searchApiKeyCredential == null || StringUtils.isBlank(searchApiKeyCredential.getApiKey())) {
```

Comment:
`null` should throw `NullPOinterException` and empty string should throw `IllegalArgumentException`. See [Java Guidelines](https://azure.github.io/azure-sdk/java_implementation.html#error-handling)